### PR TITLE
Add a limit on max DNS batch size

### PR DESCRIPTION
**Pull request by alistairking**
Otherwise we can exceed max payload sizes at ingest.

### CODE REVIEW

File: src/mode/dns.rs [hunk 1]
```diff
@@ -157,20 +159,21 @@ impl Dns {
...
-            let timeout = Duration::milliseconds(10).unsigned_abs();
-            let len = rs.len();
-            match self.client.send(rs, timeout) {
-                Ok(..) => debug!("DNS batch sent: {}", len),
-                Err(e) => warn!("DNS queue full: {:?}", e),
-            };
-
-            self.asm.flush(ts);
-            self.last = ts;
+        if (ts - self.last) < Duration::seconds(1) || self.buffer.len() < MAX_BUFFER_LEN {
```
Comment (jabersmith) [inline src/mode/dns.rs:162]: Shouldn't this be &&, not ||?  We want to send the buffer if either time or space limit is exceeded, so return if we're still inside both limits?

Comment (alistairking) [reply src/mode/dns.rs]: Yes indeed!

**Review by jabersmith — APPROVED** [review]
works for me

**Comment by kentik-will** [comment]
Thank you! I think doing this batching at the DNS API client is a better choice and I've implemented that here: https://github.com/kentik/kentik-api-rust/commit/8fc4098f9ce376b5894fb4cffb46f2d070cc06d4.

**Pull request closed without merging**